src/mars.c

```diff
@@ -1193,6 +1194,7 @@ Language changes listed by -transition=id:\n\
         VersionCondition::addPredefinedGlobalIdent("D_NoBoundsChecks");
 
     VersionCondition::addPredefinedGlobalIdent("D_HardFloat");
+    objc_tryMain_dObjc();
```

Comment:
You can segregate things properly by implementing a namespace (via a struct).  Look at ``port.c`` or ``target.c`` as an example.  The header is the shared implementation between front-ends, and the source is managed by each compiler vendor.